projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -308,6 +308,55 @@ private static function register_contact_form_blocks() {
 		Contact_Form_Block::register_child_blocks();
 	}
 
+	/**
+	 * Build the CSS for the child layout.
+	 * This replicates part of the `wp_render_layout_support_flag` function from WordPress core.
+	 * That function doesn't work for form fields due to the way they're rendered as shortcodes,
+	 * while the core function will only generate and apply classnames for layout to html.
+	 *
+	 * @param array $layout - the block's `style.layout` attribute.
+	 *
+	 * @return string CSS for the child layout.
+	 */
+	private static function build_child_layout_css( $layout ) {
+		$css = '';
+
+		if ( ! $layout ) {
+			return $css;
+		}
+
+		$self_stretch = isset( $layout['selfStretch'] ) ? $layout['selfStretch'] : null;
+
+		if ( 'fixed' === $self_stretch && isset( $layout['flexSize'] ) ) {
+			$css .= 'flex-basis: ' . $layout['flexSize'] . ';';
+			$css .= 'box-sizing: border-box;';
+		} elseif ( 'fill' === $self_stretch ) {
+			$css .= 'flex-grow: 1;';
+		}
+
+		$column_start = isset( $layout['columnStart'] ) ? $layout['columnStart'] : null;
```

Comment:
We shouldn't need to deal with `columnStart` or `rowStart` here unless we're explicitly supporting the experimental grid features in gutenberg. _If_ we're doing that, we should also include the responsive logic that unsets them at small breakpoints. I think on the whole it'd be simpler to just not support it (given we have no idea when those features will be stable)

Reply:
Nice, that makes it much simpler, thanks. 👍